addons/mail/static/src/js/tools/debug_manager.js

```diff
@@ -1,33 +1,45 @@
-odoo.define('mail.DebugManager.Backend', function (require) {
-"use strict";
+/** @odoo-module **/
 
-var core = require('web.core');
-var DebugManager = require('web.DebugManager.Backend');
+import { patch } from 'web.utils';
+import { useDebugManager } from "@web/debug/debug_manager";
+import { ViewAdapter } from "@web/legacy/action_adapters";
 
-var _t = core._t;
-/**
- * adds a new method available for the debug manager, called by the "Manage Messages" button.
- *
- */
-DebugManager.include({
-    getMailMessages: function () {
-        var selectedIDs = this._controller.getSelectedIds();
-        if (!selectedIDs.length) {
-            console.warn(_t("No message available"));
-            return;
-        }
-        this.do_action({
-            res_model: 'mail.message',
-            name: _t('Manage Messages'),
-            views: [[false, 'list'], [false, 'form']],
-            type: 'ir.actions.act_window',
-            domain: [['res_id', '=', selectedIDs[0]], ['model', '=', this._controller.modelName]],
-            context: {
-                default_res_model: this._controller.modelName,
-                default_res_id: selectedIDs[0],
-            },
-        });
-    },
+patch(ViewAdapter.prototype, 'mail.view_adapter', {
+    setup() {
+        const envWowl = this.env;
+        useDebugManager((accessRights) =>
+            setupDebugMailMessage(envWowl, this, this.props.viewParams.action)
+        );
+        this._super(...arguments);
+    }
 });
```

Comment:
Don't patch to add elements to the debug manager. And don't do useDebugManager as a way in. 
Use the simpler `import { debugRegistry } from "@web/debug/debug_registry";`
```js
debugRegistry.add("mail.manage_messages", debugMailMessage);
```

Reply:
@SimonGenin I wanted to call ```getSelectedIds``` here, do we have a alternative to call it any other way?

Reply:
I'll ask my team tomorrow

Reply:
@msh-odoo So we indeed didn't have a nice way around this. So in a way, your patch was one of the possible solution. We are trying to give the debugmanager a second registry. One for global items and one for more specific ones. 
You can see a wip here https://github.com/odoo-dev/odoo/pull/738